app/data/key_store.go

```diff
@@ -1,12 +1,32 @@
 package data
 
 import (
-	"crypto/rsa"
+	"github.com/keratin/authn-server/app/data/private"
+	"gopkg.in/square/go-jose.v2"
 )
 
 type KeyStore interface {
 	// Returns the current key
-	Key() *rsa.PrivateKey
+	Key() *private.Key
 	// Returns recent keys (including current key)
-	Keys() []*rsa.PrivateKey
+	Keys() []*private.Key
+}
+
+// Implements authn-go JWKProvider
```

Comment:
This may fit better in authn-go, since authn-server has no reason to know about the interface.

If it stays here, though, I'd ask to move it into a new file.

Reply:
Agreed on move... I went to add this into authn-go, but I thought it was kind of nice that authn-go doesn't depend directly on a particular version of authn-server, which it would do if we import `KeyStore`. It means they won't get into a dependency fight for a module that depends on both, though it seems unlikely that you wouldn't want to upgrade one or the other to get around that.

If you prefer a hard dependency then I can move it, but for now i've put it in its own file

Reply:
Ahh, I see. This is probably the better compromise.

There's a third option though: leave the code in an integration layer that already depends on both authn-go and authn-server. In this case it would be your app. One downside is that no one else would have access to it.

Can you think of other downsides?

Do you expect more integration code like this?

Reply:
I think the third way makes sense. There probably will be some more, but we could always factor that out into some kind of 'lib-keratin' once we can prove they make sense. So happy to factor this out now if you'd prefer.